crypten/mpc/mpc.py

```diff
@@ -218,6 +231,111 @@ def to(self, *args, **kwargs):
             self.share = share
             return self
 
+    def add(self, y):
+        return self.clone().add_(y)
+
+    def add_(self, y):
+        """added addition according to MAC"""
+
+        if self._mac is not None:
+            private = isinstance(y, MPCTensor)
+
+            if not private:
+                self._tensor += y
+                scaled_y = self._tensor.encoder.encode(y)
+                self._mac += self.alpha * scaled_y
+            else:
+                self._tensor += y._tensor
+                self._mac += y._mac
+
+        return self
+
+    def mul(self, y):
+        return self.clone().mul_(y)
+
+    def mul_(self, y):
+        """added multiplication according to MAC"""
+
+        if self._mac is not None:
+            private = isinstance(y, MPCTensor)
+
+            if not private:
+                self._tensor *= y
```

Comment:
This line should happen even if `self._mac is None`

Reply:
Ok, is it going to be
```
if self._mac is not None:
    self._mac *= scaled_y
```
And put all other code outside of the if statement?

Reply:
So, I think we can do the following:

```
if self._mac is None:
    if isinstance(y, MPCTensor):
        y = y._tensor
    self._tensor = self._tensor.mul_(y)
```

This will take care of the private multiplication case for passive security.

(Note: this code is taken from `_add_inplace_binary_passthrough_function` on line 1394)